cmake/MiscCheck.cmake

```diff
@@ -323,6 +323,15 @@ if (USE_ASAN)
   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${CAFFE2_ASAN_FLAG}")
 endif()
 
+if (USE_TSAN)
+  set(CAFFE2_ASAN_FLAG "-fsanitize=thread -fPIE -pie")
```

Comment:
Nit: Do you want to have this be `CAFFE2_TSAN_FLAG` (instead of CAFFE2_ASAN_FLAG) for clarity and use it below?

Reply:
Updated.